tests/syft/core/store/memory_storage_test.py

```diff
@@ -1,39 +1,164 @@
-#
-#     assert store.get_objects_of_type(np.array) == [data1, data2]
-#     assert list(store.keys()) == [k1, k2]
-#     assert list(store.values()) == [data1, data2]
-#
-#     store.clear()
-#
-#     assert len(store) == 0
-#     assert store.__sizeof__() == 48
+"""In this test suite, we evaluate the MemoryStore class. For more info
+on the MemoryStore class and its purpose, please see the documentation
+in the class itself.
+
+Table of Contents:
+    - INITIALIZATION: tests for ways MemoryStore can be initialized
+    - CLASS METHODS: tests for the use of MemoryStore's class methods
+"""
+
+import torch as th
+import re
+
+from syft.core.store.storeable_object import StorableObject
+from syft.core.store.store_memory import MemoryStore
+from syft.core.common import UID
+from syft.core.store import ObjectStore
+from syft.core.common.object import ObjectWithID
+
+from typing import Tuple, List
+
+
+def generate_id_obj(
+    data: th.Tensor, description: str, tags: List[str]
+) -> Tuple[UID, StorableObject]:
+    id = UID()
+    obj = StorableObject(id=id, data=data, description=description, tags=tags)
+
+    return id, obj
+
+
+# --------------------- INITIALIZATION ---------------------
+
+
+def test_create_memory_storage() -> None:
+    """Test that creating MemoryStore() does in fact create
+    an ObjectStore."""
+
+    store = MemoryStore()
+    assert isinstance(store, ObjectStore)
+
+
+# --------------------- CLASS METHODS ---------------------
+
+
+def test_set_item() -> None:
+    """Tests that __setitem__ and __getitem__ work intuitively."""
+
+    store = MemoryStore()
+    id1, obj1 = generate_id_obj(
+        data=th.Tensor([1, 2, 3, 4]),
+        description="Dummy tensor",
+        tags=["dummy", "tensor"],
+    )
+
+    store[id1] = obj1
+    assert id1 in store
+    assert len(store) == 1
+
+
+def test_store_item() -> None:
+    """Tests that store() works as an alternative to __setitem__."""
+
+    store = MemoryStore()
+    id1, obj1 = generate_id_obj(
+        data=th.Tensor([1, 2, 3, 4]),
+        description="Dummy tensor",
+        tags=["dummy", "tensor"],
+    )
+
+    store.store(obj=obj1)
+    assert id1 in store
+    assert len(store) == 1
+
+
+def test_get_objects_of_type() -> None:
+    """Tests that get_objects_of_type() allows MemoryStore to filter data
+    types."""
+
+    store = MemoryStore()
+    id1, obj1 = generate_id_obj(
+        data=th.Tensor([1, 2, 3, 4]),
+        description="Dummy tensor",
+        tags=["dummy", "tensor"],
+    )
+    id2, obj2 = generate_id_obj(
+        data=th.Tensor([1, 2, 3]),
+        description="Another dummy tensor",
+        tags=["another", "dummy", "tensor"],
+    )
+    id3, obj3 = generate_id_obj(
+        data=ObjectWithID(),
+        description="Dummy object with ID",
+        tags=["dummy", "object", "with", "id"],
+    )
+
+    store[id1] = obj1
+    store[id2] = obj2
+    store[id3] = obj3
+    assert store.get_objects_of_type(th.Tensor) == [obj1, obj2]
```

Comment:
Q: It might be worth converting those to ```sets```? do we impose that ```get_object_of_type``` will always return the items in the same order we added?

Reply:
Converted them to sets as they are only used in node where we can easily convert them to lists when needed.